lib/resource-mapper.js

```diff
@@ -91,22 +91,28 @@ class ResourceMapper {
     if (filePath.indexOf('/..') >= 0) {
       throw new Error('Disallowed /.. segment in URL')
     }
-    let isIndex = searchIndex && filePath.endsWith('/')
+    let isFolder = filePath.endsWith('/')
+    let isIndex = searchIndex && isFolder
 
     // Create the path for a new file
     let path
     if (createIfNotExists) {
       path = filePath
-      // Append index filename if needed
-      if (isIndex) {
-        if (contentType !== this._indexContentType) {
-          throw new Error(`Index file needs to have ${this._indexContentType} as content type`)
-        }
-        path += this._indexFilename
-      }
-      // If the extension is not correct for the content type, append the correct extension
-      if (searchIndex && this._getContentTypeByExtension(path) !== contentType) {
-        path += `$${contentType in extensions ? `.${extensions[contentType][0]}` : '.unknown'}`
+      // create a new folder
+      if (!searchIndex && isFolder) {
+  	  // else create a new file
+      } else {
+	      // Append index filename if needed
+	      if (isIndex) {
```

Comment:
Since `isIndex` only appears once (and I find its name confusing), I suggest to replace this by `searchIndex && isFolder`

Reply:
OK